parsers/ENTSOE.py

```diff
@@ -1161,26 +1205,95 @@ def _fetch_and_parse_exchange(
                 message=f"No exchange data found for {domain1} -> {domain2}",
                 zone_key=sorted_zone_keys,
             )
-        return parse_exchange(
-            raw_exchange,
-            is_import=is_import,
-            sorted_zone_keys=sorted_zone_keys,
-            logger=logger,
-            is_forecast=forecast,
-        )
+        return raw_exchange
 
     # Grab all exchanges
     for domain_pair in exchanges_to_fetch:
         # First we try to get the import data
-        raw_exchange_lists.append(
-            _fetch_and_parse_exchange(domain_pair, is_import=True)
-        )
+        raw_exchange = _fetch_exchange(domain_pair, is_import=True)
+        if raw_exchange is None:
+            raise ParserException(
+                parser="ENTSOE.py",
+                message=f"No exchange data found for {domain_pair[0]} -> {domain_pair[1]}",
+                zone_key=sorted_zone_keys,
+            )
+        if not forecast:
+            raw_exchange_lists.append(
+                parse_exchange(
+                    raw_exchange,
+                    is_import=True,
+                    sorted_zone_keys=sorted_zone_keys,
+                    logger=logger,
+                )
+            )
+        else:
+            raw_exchange_lists_forecast_day_ahead.append(
+                parse_exchange_forecast(
+                    raw_exchange,
+                    is_import=True,
+                    sorted_zone_keys=sorted_zone_keys,
+                    logger=logger,
+                    market_type=EntsoeTypeEnum.DAY_AHEAD,
+                )
+            )
+            raw_exchange_lists_forecast_total.append(
+                parse_exchange_forecast(
+                    raw_exchange,
+                    is_import=True,
+                    sorted_zone_keys=sorted_zone_keys,
+                    logger=logger,
+                    market_type=EntsoeTypeEnum.TOTAL,
+                )
+            )
+
         # Then we try to get the export data
-        raw_exchange_lists.append(
-            _fetch_and_parse_exchange(domain_pair, is_import=False)
-        )
+        raw_exchange = _fetch_exchange(domain_pair, is_import=False)
 
-    return ExchangeList(logger).merge_exchanges(raw_exchange_lists, logger)
+        if raw_exchange is None:
```

Comment:
Can't we separate more clearly the forecast and non forecasts cases?

Reply:
We can but it would likely result in more code duplication, if that's okay I can do right ahead and split them apart further.